docker/push_graphstorm_image.sh

```diff
@@ -0,0 +1,140 @@
+-a, --account       AWS Account ID. By default will get from aws-cli configuration.
+
+Example:
+
+    bash $(basename "${BASH_SOURCE[0]}") -e sagemaker --device cpu --account 123456789012 --region us-east-1
+    # Will push an image to '123456789012.dkr.ecr.us-east-1.amazonaws.com/graphstorm:sagemaker-cpu'
+
+EOF
+    exit
+}
+
+msg() {
+    echo >&2 -e "${1-}"
+}
+
+die() {
+    local msg=$1
+    local code=${2-1} # default exit status 1
+    msg "$msg"
+    exit "$code"
+}
+
+parse_params() {
+    # default values of variables set from params
+    DEVICE_TYPE="gpu"
+    GSF_HOME="${SCRIPT_DIR}/../"
+    IMAGE_NAME='graphstorm'
+    SUFFIX=""
+    REGION=$(aws configure get region) || REGION=""
+    REGION=${REGION:-us-east=1}
+    ACCOUNT=$(aws sts get-caller-identity --query Account --output text)
+
+    while :; do
+        case "${1-}" in
+        -h | --help) usage ;;
+        -x | --verbose) set -x ;;
+        --no-color) NO_COLOR=1 ;;
+        -e | --environment)
+            EXEC_ENV="${2-}"
+            shift
+            ;;
+        -i | --image)
+            IMAGE_NAME="${2-}"
+            shift
+            ;;
+        -d | --device)
+            DEVICE_TYPE="${2-}"
+            shift
+            ;;
+        -s | --suffix)
+            SUFFIX="${2-}"
+            shift
+            ;;
+        -r | --region)
+            REGION="${2-}"
+            shift
+            ;;
+        -a | --account)
+            ACCOUNT="${2-}"
+            shift
+            ;;
+        -?*) die "Unknown option: $1" ;;
+        *) break ;;
+        esac
+        shift
+    done
+
+    [[ -z "${EXEC_ENV-}" ]] && die "Missing required parameter: -e/--environment [local|sagemaker]"
+
+    return 0
+}
+
+cleanup() {
+    trap - SIGINT SIGTERM ERR EXIT
+    # script cleanup here
+}
+
+parse_params "${@}"
+
+if [[ ${EXEC_ENV} == "sagemaker" || ${EXEC_ENV} == "local" ]]; then
+    : # Do nothing
+else
+    die "--environment parameter needs to be one of 'sagemaker', or 'local' got ${EXEC_ENV}"
+fi
+
+TAG="${EXEC_ENV}-${DEVICE_TYPE}${SUFFIX}"
+LATEST_TAG="${EXEC_ENV}-${DEVICE}latest"
+IMAGE="${IMAGE_NAME}"
+
+msg "Execution parameters: "
+msg "- ENVIRONMENT: ${EXEC_ENV}"
+msg "- DEVICE TYPE: ${DEVICE_TYPE}"
+msg "- IMAGE: ${IMAGE}"
+msg "- TAG: ${TAG}"
+msg "- REGION: ${REGION}"
+msg "- ACCOUNT: ${ACCOUNT}"
+
+FULLNAME="${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/${IMAGE}:${TAG}"
+LATEST_FULLNAME="${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/${IMAGE}:${LATEST_TAG}"
+
+# If the repository doesn't exist in ECR, create it.
+echo "Getting or creating container repository: ${IMAGE}"
+if ! eval aws ecr describe-repositories --repository-names "${IMAGE}" --region ${REGION} >/dev/null 2>&1; then
+    msg "WARNING: ECR repository ${IMAGE} does not exist in region ${REGION}. Creating..."
+    aws ecr create-repository --repository-name "${IMAGE}" --region ${REGION} >/dev/null
```

Comment:
Can we catch the error case when the user's account does not have the permision to create a new ECR?